Updated code to be compliant with Zig 0.12

7Zifle opened the pull request:
This pull request makes necessary changes to get the codebase compliant with Zig 0.12. The test.Dockerfile builds musl 1.2.4 because Alpine Linux is using musl 1.2.5 which is incompatible with Zig 0.12. I wasn't able to find a simpler way of getting musl 1.2.4 in the testing environment.

7Zifle committed:
Updated code to be compliant with zig-0.12

7Zifle committed:
Fixed build issues in test.Dockerfile

fkollmann commented:
Thanks for the PR! I just realized the new Zig 0.12 got released. Please give me a few days to work through the changes (both this PR and Zig).

PS: Saw your previous PR. GitHub will automatically update a PR, if you push new changes.
PPS: Actually `musl-dev` in Alpine 3.19 is v1.2.4 (https://pkgs.alpinelinux.org/package/v3.19/main/x86_64/musl-dev).

7Zifle committed:
Removed useless build of musl and updated examples

- Zig 0.12 package was moved from Alpine's Edge Testing to Edge
  Community

7Zifle commented:
Thank you for following up on this. I've updated the `test.Dockerfile` so it no longer includes building `musl`. It appears the problem I was having with the docker unit tests was related to a dependency on clang. I also went ahead and updated the examples `build.zig` files. Just a heads up, The `build.zon` files are still using `zzmq-v0.2.1-zig0.11`

7Zifle committed:
Fixed errors in examples from Zig 0.12 update

- Errors with `stopRunning` in dealer examples resolved
- Errors relating to signal handling in dealer examples resolved

7Zifle commented:
I apologize for the number of commits in this update. While reading over the examples, I noticed that several standard library functions referenced have been either relocated or modified slightly.

fkollmann commented:
FYI, since the PR contains various aspects, I will merge it into a new `pr3_stage` branch and work out the details from there.
